### DISCUSSION TITLE
litellm as service

### DISCUSSION THREAD

**Discussion by sfiodorov** [BerriAI/litellm#1048] [Q&A] [answered]
Hello, we are currently searching for an AI abstraction layer that includes features such as user management and rate limiting. We have heard that Litellm is a great solution for our needs. However, our team is not familiar with the Python stack.  I was wondering if these same features can be accessed via a Docker container?
If not, if there is a SAAS option?

**Comment by ghost** [comment]
Hey @sfiodorov 

Does the docker image not work? https://github.com/BerriAI/litellm/blob/main/Dockerfile

Here's the docs: https://docs.litellm.ai/docs/proxy/quick_start#quick-start-docker-image-github-container-registry

If you're looking for a hosted solution, I'd recommend checking out Helicone

**Comment by ishaan-jaff** [comment]
Hi @sfiodorov we'd love to help, sharing a link to our calendar if you'd like to meet and discuss this: https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat

**Reply by sfiodorov** [reply]
Hi.
Thanks for answering.
Im not sure if the docker version contains the same set of features as a library.
For example it seems that there is no client-level budget management and rate limiting.

**Reply by ghost** [reply]
@sfiodorov the docker just builds the library - https://github.com/BerriAI/litellm/blob/ed07012874a319cbc95c4e99a332973918746de9/Dockerfile#L16

Here's how to track spend/user - https://docs.litellm.ai/docs/proxy/virtual_keys

a call would help us get context

**Reply by sfiodorov** [reply]
Thanks.
Track spend is not what we are looking for.
We need to limit client rates and budgets. Something like here: https://docs.litellm.ai/docs/budget_manager
But yes, it is better to have a dedicated call. I will check with my team and come back.
Thanks again for quick response

**Reply by ghost** [reply, marked as answer]
Hey @sfiodorov  we now enable client budget limits - https://docs.litellm.ai/docs/proxy/virtual_keys#set-budgets

Let me know if this is what you were looking for. Also happy to hop on a call to discuss. 

Attaching my socials so we can set up a dedicated support channel for your team: 
- LinkedIn: https://www.linkedin.com/in/krish-d/
- Discord: Krrish#8748

**Comment by sfiodorov** [comment]
Hey, that sounds good.
We will evaluate it ASAP.
Thanks for letting us know.
